shared/video-sdk/develop/speech-to-text/project-implementation/android.mdx

```diff
@@ -0,0 +1,372 @@
+                            showMessage("Speech to text status: " + status + "");
+                            runOnUiThread(() -> btnSpeechToText.setText("Start speech to text"));
+                        }
+                    } else {
+                        String message = jsonObject.getString("message");
+                        showMessage("Exception querying speech-to-text task: " + message);
+                    }
+
+                } catch (Exception e) {
+                    Log.i("Error parsing query call response", result);
+                }
+            }
+        }
+        );
+    }
+    ```
+
+1. **Stop the task**
+
+    To [stop speech to text](#stop-speech-to-text), you send a `DELETE` request with the, `appId`, `taskId` and `builderToken` in the URL. In the `MainActivity` class, add the following method:
+
+    ```java
+    public void stopSpeechToText() {
+        String url = baseUrl + "/v1/projects/" + appId
+                + "/rtsc/speech-to-text/tasks/" + speechToTextTaskId
+                + "?builderToken=" + builderToken;
+
+        OkHttpClient httpClient = new OkHttpClient().newBuilder().build();
+        MediaType mediaType = MediaType.parse("application/json");
+        Request request = new Request.Builder()
+                .addHeader("Content-Type", "application/json")
+                .addHeader("Authorization", authorizationHeader)
+                .url(url)
+                .delete()
+                .build();
+
+        httpClient.newCall(request).enqueue(new Callback() {
+            @Override
+            public void onFailure(final Call call, IOException e) {
+                showMessage(e.toString());
+            }
+
+            @Override
+            public void onResponse(Call call, final Response response) throws IOException {
+                String result = response.body().string();
+                if (response.isSuccessful()) {
+                    showMessage("Speech-to-text task stopped");
+                    speechToTextTaskId = "";
+                    isSpeechToTextRunning = false;
+                    runOnUiThread(() -> btnSpeechToText.setText("Start speech to text"));
+                } else {
+                    showMessage("Exception stopping speech-to-text task: " + result);
+                }
+            }
+        }
+        );
+    }
+    ```
+
+1. **Receive and display the text stream**
+
+    When you set one of the `destinations` in the starting configuration as `AgoraRTCDataStream`, the speech-to-text conversion output stream is pushed to your RTC channel. To receive this stream and display the text in your <Vpl k="CLIENT" />, add the following code after `private final IRtcEngineEventHandler mRtcEventHandler = new IRtcEngineEventHandler() {` in the `MainActivity` class:
+
+    ```java
+    @Override
+    public void onStreamMessage(int uid, int streamId, byte[] data) {
+        showMessage(data.toString());
+    }    
```

Comment:
Getting an error `Method does not override method from its superclass` here. My build.gradle has `implementation 'io.agora.rtc:full-sdk:4.0.1'`.

Reply:
You need to use the 4.1.0 SDK. The callback is probably not implemented in 4.0.1.